common/src/main/java/com/viaversion/viaversion/protocols/protocol1_9to1_8/types/ChunkBulk1_8Type.java

```diff
@@ -0,0 +1,122 @@
+import com.viaversion.viaversion.api.type.PartialType;
+import com.viaversion.viaversion.api.type.Type;
+import com.viaversion.viaversion.api.type.types.minecraft.BaseChunkBulkType;
+import com.viaversion.viaversion.protocols.protocol1_9_3to1_9_1_2.storage.ClientWorld;
+import io.netty.buffer.ByteBuf;
+
+public class ChunkBulk1_8Type extends PartialType<Chunk[], ClientWorld> {
+
+    public ChunkBulk1_8Type(final ClientWorld clientWorld) {
+        super(clientWorld, Chunk[].class);
+    }
+
+    @Override
+    public Class<? extends Type> getBaseClass() {
+        return BaseChunkBulkType.class;
+    }
+
+    @Override
+    public Chunk[] read(ByteBuf input, ClientWorld world) throws Exception {
+        final boolean skyLight = input.readBoolean();
+        final int count = Type.VAR_INT.readPrimitive(input);
+        final Chunk[] chunks = new Chunk[count];
+        final ChunkBulkSection[] chunkInfo = new ChunkBulkSection[count];
+
+        // Read metadata
+        for (int i = 0; i < chunkInfo.length; i++) {
+            chunkInfo[i] = new ChunkBulkSection(input, skyLight);
+        }
+        // Read data
+        for (int i = 0; i < chunks.length; i++) {
+            final ChunkBulkSection chunkBulkSection = chunkInfo[i];
+            chunkBulkSection.readData(input);
+            chunks[i] = Chunk1_8Type.deserialize(chunkBulkSection.chunkX, chunkBulkSection.chunkZ, true, skyLight, chunkBulkSection.bitmask, chunkBulkSection.getData());
+        }
+
+        return chunks;
+    }
+
+    @Override
+    public void write(ByteBuf output, ClientWorld world, Chunk[] chunks) throws Exception {
+        boolean skyLight = false;
+        for (Chunk c : chunks) {
+            for (ChunkSection section : c.getSections()) {
+                if (section != null) {
+                    if (section.getLight().hasSkyLight()) {
+                        skyLight = true;
+                    }
+                }
+            }
+        }
+        output.writeBoolean(skyLight);
+        Type.VAR_INT.writePrimitive(output, chunks.length);
+
+        // Write metadata
+        for (Chunk c : chunks) {
+            output.writeInt(c.getX());
+            output.writeInt(c.getZ());
+            output.writeShort(c.getBitmask());
+        }
+        // Write data
+        for (Chunk c : chunks) {
+            output.writeBytes(Chunk1_8Type.serialize(c));
+        }
+    }
+
+    public static final class ChunkBulkSection {
+        private final int chunkX;
+        private final int chunkZ;
+        private final int bitmask;
+        private final byte[] data;
+
+        public ChunkBulkSection(final ByteBuf input, final boolean skyLight) {
+            this.chunkX = input.readInt();
+            this.chunkZ = input.readInt();
+            this.bitmask = input.readUnsignedShort();
+            final int bitCount = Integer.bitCount(this.bitmask);
+            this.data = new byte[(bitCount * ((4096 * 2) + 2048)) + (skyLight ? bitCount * 2048 : 0) + 256];
```

Comment:
Make comments explaining the numbers/put them into private static final fields

Reply:
Those numbers are from the previous chunk reading code, idk what they do exactly